### PR TITLE
Improve build speeds for pull requests

### PR DESCRIPTION
**Pull request by davidrapson**
This PR does a couple of things to improve build speeds for pull requests:

- Updates unit tests and cypress tests to run in parallel
- Merges cypress tests suites to into either common or e2e to avoid extra boot up time between specs
- Switch to use travis build stages in order to avoid unnecessary steps when not deploying (particularly the `npm prune` stage

In good cases testing down from 7-8 minutes to 5-6 minutes. I do still see travis take ~2minutes to even reach the install stage so I'm not convinced we can shave _that_ much time off overall but when the wind is behind the build these changes should help.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,65 +1,77 @@
...
-node_js:
-- '8'
-before_install:
-- npm i -g npm@6
-install:
-- npm ci
+node_js: '8'
+
+# Install latest npm & dependencies
+before_install: npm i -g npm@6
```
Comment (mattandrews) [inline .travis.yml]: Does the 6 here conflict with the 8 above?

File: cypress/integration/integration_spec.js [hunk 2]
```diff
@@ -62,10 +242,10 @@ describe('e2e', function() {
         cy.get('#js-mobile-nav-toggle').as('navToggle');
         cy.get('#qa-offscreen-navigation').as('nav');
 
-        cy.get('@navToggle').click();
+        cy.get('@navToggle').click({ force: true });
```
Comment (mattandrews) [inline cypress/integration/integration_spec.js]: What's this doing (and why)?

Comment (davidrapson) [reply cypress/integration/integration_spec.js]: Oh good point, there was something funny going on with the scroll position when I upgraded Cypress, so this bypasses the visibility check. This was a workaround. It's not a great one though, so let me double check.

File: .travis.yml (hunk 1 above)
Comment (davidrapson) [reply .travis.yml]: Nope, this is **npm** 6. Node 8 comes with npm 5 but we want 6 for `npm ci`. This hasn't changed from the previous config. 

This is all moot if we upgrade to Node 10 as in this PR https://github.com/biglotteryfund/blf-alpha/pull/1162 as Node 10 is bundled with npm 6.

Comment (mattandrews) [reply .travis.yml]: This is what I get for doing code reviews before coffee. Carry on!

File: cypress/integration/integration_spec.js (hunk 2 above)
Comment (mattandrews) [reply cypress/integration/integration_spec.js]: I don't mind the code, just wasn't clear what it was doing. Add a comment maybe?

Comment (davidrapson) [reply cypress/integration/integration_spec.js]: If I can't find a better solution then will do

**Pull request merged**